ksqldb-common/src/main/java/io/confluent/ksql/util/DecimalUtil.java

```diff
@@ -258,7 +259,7 @@ private static void ensureMax(final BigDecimal value, final int precision, final
     final int digits = precision - scale;
     final BigDecimal maxValue = BigDecimal.valueOf(Math.pow(10, digits));
     if (maxValue.compareTo(value.abs()) < 1) {
-      throw new ArithmeticException(
```

Comment:
This ensureMax is an utility method that I think is called in different places other than functions, not just functions. In those cases, then KsqlFunctionException won't be a valid exception.

Reply:
@spena Thanks for pointing this out. After discussing with @jnh5y we have decided to revert this and make the change locally in the sum udaf.